### PR TITLE
 fix potential NullPointerException

### PR DESCRIPTION
**Pull request by tzy1316106836**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
___
### Bug fix
- Bug description.
the inappropriate use of equals() 
- How to fix?
fix the use of equals(),fix potential NullPointerException

### CODE REVIEW

**Comment by wu-sheng** [comment]
@IanCao CI is failing mostly because Circle CI change. I and @kezhenxu94 will make it back online soon.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/GCProvider.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ private GCMetricAccessor findByBeanName(String name) {
         } else if (name.indexOf("G1") > -1) {
             // G1 collector ( -XX:+UseG1GC )
             return new G1GCModule(beans);
-        } else if (name.equals("MarkSweepCompact")) {
+        } else if ("MarkSweepCompact".equals(name)) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/GCProvider.java]: When the name is NULL?

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/match/HierarchyMatch.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ private void matchHierarchyClass(TypeDescription.Generic clazz, List<String> par
         }
 
         TypeDescription.Generic superClazz = clazz.getSuperClass();
-        if (superClazz != null && !clazz.getTypeName().equals("java.lang.Object")) {
+        if (superClazz != null && !"java.lang.Object".equals(clazz.getTypeName())) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/match/HierarchyMatch.java]: When class type name is NULL?

Comment (tzy1316106836) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/match/HierarchyMatch.java]: in this case, the class type is not null, but  equqls() was usually used in this style

**Comment by wu-sheng** [comment]
First of all, this PR is not fixing potential NPE from my current understanding. For coding style, in these cases(your changes), I could only say it is harmless.

**Pull request closed without merging**